pages/admin/typeform/index.tsx

```diff
@@ -31,8 +31,44 @@ const ApplicationsEditPage: NextPage = () => {
     },
   );
   if (!isOfficer) return <AdminOnlyComponent />;
-  if (isLoading || status == 'loading') return <p className="text-gray-100">loading...</p>;
-  if (error) return <ErrorComponent errorCode={(error as GraphQLError).extensions.code as string} errorMessage={(error as GraphQLError).message}/>;
+  const array = [];
+  for (let i = 0; i < 20; i++) {
+    array.push(
+      <div className="flex items-center justify-between pt-4">
+        <div>
+          <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24 mb-2.5"></div>
+          <div className="w-32 h-2 bg-gray-200 rounded-full dark:bg-gray-700"></div>
+        </div>
+        <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-700 w-12"></div>
+      </div>,
+    );
+  }
+  if (isLoading || status == 'loading')
+    return (
+      <>
+        <div
+          role="status"
+          className="w-full h-full p-4 space-y-4 border border-gray-200 divide-y divide-gray-200 rounded shadow animate-pulse dark:divide-gray-700 md:p-6 dark:border-gray-700"
+        >
+          <div className="flex items-center justify-between">
+            <div>
+              <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-600 w-24 mb-2.5"></div>
+              <div className="w-32 h-2 bg-gray-200 rounded-full dark:bg-gray-700"></div>
+            </div>
+            <div className="h-2.5 bg-gray-300 rounded-full dark:bg-gray-700 w-12"></div>
+          </div>
+          {array}
+          <span className="sr-only">Loading...</span>
+        </div>
+      </>
```

Comment:
Seems like this piece of code is being duplicated across multiple pages. Can you abstractify it into a component and import it?

- [ ] abstractify